plugins/samples/overwrite_header/tests.textpb

```diff
@@ -0,0 +1,37 @@
+# Expect no matches, so no overwrite.
+test {
+  name: "NoOverwrite"
```

Comment:
Note that the new headers (RequestHeader/ResponseHeader) still get added in this test. Please add the corresponding expectations to result{}.

Reply:
I did not understand. The headers mentioned by you aren't added here. So no changes are expected here.

Reply:
Ah, my point was that these lines are always called:
```
# request path
replaceRequestHeader("RequestHeader", "changed");
# response path
replaceResponseHeader("ResponseHeader", "changed");
```
Regardless of whether the input has "RequestHeader" or "ResponseHeader", this code will set those values into the output. So IMO the "NoOverwrite" test should demonstrate that the headers always get added, by verifying this in the result{} stanza:
```
    result { 
      has_header { key: "RequestHeader" value: "changed" }
    }
```

---
What you have written: `result { no_header {value: "changed"} }` doesn't test what you expect. Because the key is unspecified, it checks that there's no header called "" (always true I expect). I can fix/clarify this in the test framework.

Reply:
https://github.com/GoogleCloudPlatform/service-extensions-samples/pull/106 to fix

Reply:
Now I got it... This plugin is intended to only overwrite and not to add if its missing.
I've just made the corresponding changes to the code and tests.
Pls take a look.

Reply:
Thanks for fixing